pkg/crc/cluster/cluster.go

```diff
@@ -22,33 +21,7 @@ func WaitForSsh(sshRunner *ssh.SSHRunner) error {
 	return errors.RetryAfter(10, checkSshConnectivity, time.Second)
 }
 
-// CheckCertsValidityUsingBundleBuildTime check if the cluster certs going to expire soon.
-func CheckCertsValidityUsingBundleBuildTime(buildTime time.Time) (bool, int) {
-	certExpiryDate := buildTime.AddDate(0, 1, 0)
-	// Warn user if the cert expiry going to happen starting of the 7 days
-	timeAfter7Days := time.Now().AddDate(0, 0, 7)
-	return timeAfter7Days.After(certExpiryDate), int(time.Until(certExpiryDate).Hours()) / 24
-}
-
-// CheckCertsValidity checks if the cluster certs have expired or going to expire in next 7 days
-func CheckCertsValidity(sshRunner *ssh.SSHRunner) (bool, int, error) {
-	certExpiryDate, err := getcertExipryDateFromVM(sshRunner)
-	if err != nil {
-		return false, 0, err
-	}
-	if time.Now().After(certExpiryDate) {
-		return false, 0, fmt.Errorf("Certs have expired, they were valid till: %s", certExpiryDate.Format(time.RFC822))
-	}
-
-	// Warn user if the cert expiry going to happen starting of the 7 days
-	timeAfter7Days := time.Now().AddDate(0, 0, 7)
-	if timeAfter7Days.After(certExpiryDate) {
-		return true, int(time.Until(certExpiryDate).Hours()) / 24, nil
-	}
-	return false, 0, nil
-}
-
-func getcertExipryDateFromVM(sshRunner *ssh.SSHRunner) (time.Time, error) {
+func GetCertExpiryDateFromVM(sshRunner *ssh.SSHRunner) (time.Time, error) {
```

Comment:
not used to see us passing around the `SSHRunner` instead of `Host` in the context of VM.

Reply:
Now that we are changing the ssh key of the VM without explicit Host interaction, this SSHRunner is needed so that we know which ssh key we should be using. At some point, we probably should introduce a crc.VM type which would have host.Host and SSHRunner fields, but this does not exist at the moment.

Reply:
OK. we need to work on that abstraction sooner than later